limiter.go

```diff
@@ -22,9 +22,10 @@ type Context struct {
 
 // Limiter is the limiter instance.
 type Limiter struct {
-	Store   Store
-	Rate    Rate
-	Options Options
+	Store         Store
+	Rate          Rate
+	Options       Options
+	ErrValidation error
```

Comment:
Please remove this `ErrValidation`, if you should return an error, return it directly.

Reply:
It was the simplest way to stop the handler by calling [OnError](https://github.com/jigsaw373/limiter/blob/69474dbc4505b73fd8829f1e8a78e0c52cc2c898/drivers/middleware/stdlib/middleware.go#L58) function

Reply:
This doesn't work because either you have one request at a time, and even then it doesn't work because you never reset the `ErrValidation` for the next request.

Or, you have multiple request at the same time, and then you have a huge data race / race condition.

This is clear now that the API for `KeyGetter` is not enough but, if we change it, it would break backward compatibility. **My bad for this!**

You'll have to wait for the `v4` engine to add support for `JWT`.